osgp/protocol-adapter-iec61850/osgp-protocol-adapter-iec61850/src/main/java/org/opensmartgridplatform/adapter/protocol/iec61850/infra/networking/services/commands/Iec61850GetFirmwareVersionCommand.java

```diff
@@ -27,11 +24,19 @@
 import org.opensmartgridplatform.adapter.protocol.iec61850.services.DeviceMessageLoggingService;
 import org.opensmartgridplatform.dto.valueobjects.FirmwareModuleType;
 import org.opensmartgridplatform.dto.valueobjects.FirmwareVersionDto;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class Iec61850GetFirmwareVersionCommand {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(Iec61850GetFirmwareVersionCommand.class);
 
+    private DeviceMessageLoggingService loggingService;
+
+    public Iec61850GetFirmwareVersionCommand(final DeviceMessageLoggingService loggingService) {
+        this.loggingService = loggingService;
+    }
+
     public List<FirmwareVersionDto> getFirmwareVersionFromDevice(final Iec61850Client iec61850Client,
             final DeviceConnection deviceConnection) throws ProtocolAdapterException {
         final Function<List<FirmwareVersionDto>> function = new Function<List<FirmwareVersionDto>>() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this anonymous class number of lines from 30 to at most 20, or make it a named class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1188)